catalog/app/containers/Bucket/Listing.js

```diff
@@ -264,16 +340,129 @@ const useListingStyles = M.makeStyles((t) => ({
   },
 }))
 
-export default function Listing({ items, truncated = false, locked = false, loadMore }) {
+export function ListingWithPrefixFiltering({
+  items,
+  truncated = false,
+  locked = false,
+  prefix = '',
+  prefixValue,
+  changePrefixValue,
+  commitPrefixChange,
+}) {
   const classes = useListingStyles()
 
   const scrollRef = React.useRef(null)
-  const scroll = React.useCallback((prev) => {
-    if (prev && scrollRef.current) scrollRef.current.scrollIntoView()
-  })
+  const scroll = React.useCallback(
+    (prev) => {
+      if (prev && scrollRef.current) scrollRef.current.scrollIntoView()
+    },
+    [scrollRef],
+  )
+
+  const totalItems = React.useMemo(() => {
+    const stats = computeStats(items)
+    return stats.dirs + stats.files
+  }, [items])
+
+  const pagination = Pagination.use(items, { perPage: 25, onChange: scroll })
```

Comment:
i dont think these values should come from one config / constant value, because they should not necessarily be the same in all those cases, it's more like they are the same "by accident", and maybe we would decide to change it in one place and not in the others.
i believe sometimes it's better to have repeating code in several places then abstracting away everything based only on the fact that it looks similar, when in reality it may not be related at all.

Reply:
Sure. I thought it's not by accident, but intentionally the same